### PR TITLE
Fix bug where portal is added to dom twice if parent does forceUpdate during initial render

### PR DESCRIPTION
**Pull request by billneff79**
Found this while using preact-portal with `apollo-graphql` which does a forceUpdate during `componentDidMount` which exposed this bug.  Unit test added that reproduces the issue.  Fails with original code and passes with this PR.

### CODE REVIEW

File: src/preact-portal.js [hunk 1]
```diff
@@ -29,6 +32,8 @@ export default class Portal extends Component {
 	}
 
 	renderLayer(show=true) {
+		if (!this.isMounted) return;
```
Comment (billneff79) [inline src/preact-portal.js]: By adding the `setTimeout` to delay the `renderLayer` call from `componentDidUpdate` to ensure that it comes after `this.remote` is set by the `render` call in the initial `renderLayer`, it had the side effect where I was occasionally seeing the call from CDU occurring _after_ the call from `componentWillUnmount`.  That's why I put the failsafe here to ensure that no re-renders occur after this controlling component has been unmounted.

Comment (billneff79) [reply src/preact-portal.js]: I tried but was unable to reproduce this in a unit test.  Would love any help with that.

Comment (developit) [reply src/preact-portal.js]: Might be worth swapping this for a clearTimeout() of the timer created in CDU instead?

**Comment by ptbrowne** [comment]
I also experience the bug, what are the needs for this to be merged ?

**Comment by billneff79** [comment]
@developit can we get this merged and a new version published?

**Comment by developit** [comment]
Yup! Sorry about that.

**Comment by developit** [comment]
Released as [1.1.3](https://github.com/developit/preact-portal/releases/tag/1.1.3).

**Pull request merged**